### ISSUE TITLE
Information that enters by adding push in post does not render colors,and the length of data added in the later stage is also much shorter

### ISSUE THREAD

**Issue by blackskymoonstar** [DHTMLX/vue-gantt-demo#44]
![issue](https://user-images.githubusercontent.com/99584652/217451449-d6cc141e-010c-473e-a89a-d86a065eb5e2.png)
![issue2](https://user-images.githubusercontent.com/99584652/217452228-b626c180-65e3-4173-8ecb-c946dec925ec.png)

**Comment by blackskymoonstar** [comment]
How data in Vue is passed to the Gantt object

**Comment by gearcoded** [comment]
@blackskymoonstar ,

In this demo, the `gantt` object is available globally. To update a task, you need to use the `getTask` method, and you will get the `task` object:
https://docs.dhtmlx.com/gantt/api__gantt_gettask.html
After modifying it, you need to call the `updateTask` method:
https://docs.dhtmlx.com/gantt/api__gantt_updatetask.html

So, you need to use the following code:
```
const task = gantt.getTask(2)
task.color = "yellow"
gantt.updateTask(2)
```

If you add the following code to the `logTaskUpdate` function, you only update the dataset that is passed to Gantt:
```
      this.tasks.data.push({
        id: 3,
        text: "task #3",
        color: "yellow",
        start_date: "2025-04-04",
        duration: 3,
        progress: 0.5
      })
```
Gantt loads the data from that dataset only once after the component is loaded and the `mounted` method is called.
After Gantt loads the data from the dataset, it stores the data internally. By default, the dataset is modified, so the tasks in the dataset and the tasks in the Gantt datastore are the same objects. 
But if you add new data in that dataset, it won't be loaded into the Gantt datastore. So it will only remain in that dataset.
You need to manually load the data again by using the `parse` method:
https://docs.dhtmlx.com/gantt/api__gantt_parse.html
Or you can add a new task with the `addTask` method:
https://docs.dhtmlx.com/gantt/api__gantt_addtask.html


Another way is to implement a custom solution to monitor the dataset that is passed to the Gantt Component.

For example, you can use the following code:
```
    gantt.attachEvent("onBeforeDataRender", () => {
        this.$props.tasks.data.forEach((task) =>{
          if (!gantt.isTaskExists(task.id)){
            gantt.addTask(task)
          }
        })
    });
```

However, the `logTaskUpdate` method is used to display the changes after a task in Gantt was updated. So, it will be called each time you modify a task. Probably, it is better to try adding the data to another method.
